### DISCUSSION TITLE
Boot into ShredOS with iPXE + EFI

### DISCUSSION THREAD

**Discussion by livestrong91** [ipxe/ipxe#412] [Q&A] [answered]
Hi,
I'm trying to boot into ShredOS from my iPXE menu on my EFI clients. ShredOS is providing a .IMG 64bit for BIOS/EFI.
https://github.com/PartialVolume/shredos.x86_64

With BIOS I made it through with 

```
:shredosbios
set base-url http://xx/shredos
kernel ${base-url}/memdisk || read void
initrd ${base-url}/shredos.img
imgargs memdisk raw
boot
```

memdisk is not compatible with EFI. So I extracted the ShredOS.IMG and found the bootx64.efi in it. So my approach for EFI is:

```
:shredosuefi
set base-url http://xx/shredos
kernel ${base-url}/bootx64.efi initrd=shredos.img
initrd ${base-url}/shredos.img
boot
```

But I'm landing in the grub2 shell. So I tried to load the grub.cfg with initrd, but I'm landing in the grub2 shell as well.
I also tried to load the shredos-file (which is mentioned in the grub.cfg) out of the boot folder of the shredos.img.

So is the wrong approach to take the bootx64.efi out of the img of shredos? Is it not bootable with EFI? 
Should I use an other kernel here?

I am very grateful for any help and hints! Thx

**Comment by NiKiZe** [comment, marked as answer]
What does the img contain? is it maybe a FAT filesystem?
have you tried sanbooting it?

You should check with the one providing the image what the correct approach is,
To boot grub from iPXE in efi mode, you could (ab)use wimboot, see https://forum.ipxe.org/showthread.php?tid=10739&pid=18277#pid18277

Also try to find a way to run it in legacy without memdisk, sanboot or try to understand the image a bit more.
(memdisk is usually a hack to create a virtual disk, instead of proper load from PXE to actual code)

**Reply by livestrong91** [reply]
Thx @NiKiZe for your reply and advice!

To your questions:
The image contains:

![image](https://user-images.githubusercontent.com/51318313/125565479-d5a9695a-4108-4c99-942d-753e6a37c4a7.png)

The 2 grub.cfg contain:

![image](https://user-images.githubusercontent.com/51318313/125566011-4290c178-fec1-4660-a076-41af14a5717d.png)

I analyzed the shredos.img a bit with fdisk -lu. It looks like you are right. It is a FAT32 system.

```
fdisk -lu shredos-2020.05.012_x86-64_0.31_20210607.img

Disk shredos-2020.05.012_x86-64_0.31_20210607.img: 51 MB, 51846656 bytes, 101263 sectors
Units = sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disk label type: dos
Disk identifier: 0x00000000

                                       Device Boot      Start         End      Blocks   Id  System
shredos-2020.05.012_x86-64_0.31_20210607.img1            1263      101262       50000    c  W95 FAT32 (LBA)
```

Does it mean, that I have to use wimboot to boot with that img?

I tried sanboot once but without luck with this shredos.img. So now I am going to dive deeper into that.

Thx for your quick help!!

**Reply by NiKiZe** [reply]
shredos is just linux,
grab the boot/shredos file from the image, put it on the server, and do kernel shredos
then if needed experiment with console= and loglevel= arguments that you can find in grub

**Reply by livestrong91** [reply]
Thx 4 your help @NiKiZe!

During my analyse of the img, I recognized that the file shredos was corrupt or not copied well. I copied the boot folder again and now it works for BIOS and UEFI. 

my ipxe menu:

```
:shredos
set base-url http://xx/shredos
kernel ${base-url}/boot/shredos console=tty3 loglevel=3
boot
```

Thank you for your quick hints and answers!!
